Add certificate management with UniFi OS integration

lox opened the pull request:
This PR adds support for generating and managing Tailscale HTTPS certificates on UniFi Dream Machines.

## Features
- Generate certificates using `tailscale cert` command
- Install certificates into UniFi OS with proper database registration for persistence
- Automatic weekly renewal via systemd timer
- Support for both UniFi OS 1.x and 2.x

## Usage
```bash
# Generate a certificate
/data/tailscale/manage.sh cert generate [hostname]

# Install into UniFi OS
/data/tailscale/manage.sh cert install-unifi [hostname]

# Restart UniFi Core
systemctl restart unifi-core
```

## Testing
Includes comprehensive test coverage for all certificate operations.

Tested on UDM Pro with UniFi OS 2.x.

lox committed:
feat: add certificate management with UniFi OS integration

- Add cert command to generate, renew, list certificates using tailscale cert
- Add install-unifi command for UniFi OS 2.x+ certificate installation
- Register certificates in PostgreSQL database for persistence across restarts
- Support UUID-based certificate system used by UniFi OS
- Add automatic certificate renewal via systemd timer (weekly)
- Clean up renewal timer on uninstall
- Add comprehensive test coverage for all certificate operations
- Add documentation to README

The database registration is essential for UniFi OS 2.x+ as certificates
are stored in PostgreSQL and must be properly registered to persist.

notheotherben commented on package/tailscale-cert-renewal.timer:
```diff
@@ -0,0 +1,10 @@
+
+[Timer]
+OnCalendar=weekly
+Persistent=true
```
Any reason you haven't specified the `Unit=tailscale-cert-renewal.service` here (in the same way we're doing in the `tailscale-install.timer`)?

notheotherben commented on package/unios_2.x.sh:
```diff
@@ -117,4 +117,186 @@ _tailscale_uninstall() {
+            mkdir -p "$cert_dir"
+            echo "Generating certificate for $hostname..."
+            
+            if tailscale cert --cert-file "$cert_dir/$hostname.crt" --key-file "$cert_dir/$hostname.key" "$hostname"; then
```
I wonder if we can use `tailscale status --peers=false --json | jq -r '.Self.DNSName[:-1]'` to get the hostname automatically, since that is the only valid input anyway.

lox replied:
Done! Used `sed 's/\.$//'` instead of `[:-1]` for better shell compatibility, but same result.

notheotherben commented on package/tailscale-cert-renewal.service:
```diff
@@ -0,0 +1,9 @@
+
+[Service]
+Type=oneshot
+ExecStart=/bin/sh -c 'for cert in /data/tailscale/certs/*.crt; do [ -f "$cert" ] && hostname="${cert##*/}" && hostname="${hostname%.crt}" && /data/tailscale/manage.sh cert renew "$hostname"; done'
```
Is this necessary? There should only be one valid certificate since only one hostname is valid - if we derive the hostname automatically using `tailscale status --peers=false --json | jq -r '.Self.DNSName[:-1]'` within the `manage.sh cert` subcommand then we can just run `manage.sh cert renew` here without needing to specify arguments.

lox replied:
You're right, simplified to just `/data/tailscale/manage.sh cert renew` 👍

notheotherben reviewed (commented):
Awesome addition, thank you for putting this together - had a look through it and the main suggestion I've got is to remove the ability to specify the `hostname` and instead derive it from `tailscale status --json` output automatically - which should simplify several aspects of the implementation and reduce the likelihood of misconfiguration/unexpected errors.

lox committed:
refactor: remove hostname parameter from cert commands

Per PR feedback, hostname is now automatically derived from
'tailscale status --json' output. This simplifies the interface
and reduces the chance of misconfiguration.

Changes:
- Remove hostname parameter from all cert commands
- Derive hostname using: tailscale status --json | jq -r '.Self.DNSName'
- Update systemd renewal service to use simplified command
- Add Unit= to timer file to match other timers in the project
- Update all tests to work with automatic hostname detection
- Update documentation to reflect the changes

lox commented:
> Awesome addition, thank you for putting this together - had a look through it and the main suggestion I've got is to remove the ability to specify the `hostname` and instead derive it from `tailscale status --json` output automatically - which should simplify several aspects of the implementation and reduce the likelihood of misconfiguration/unexpected errors.

Thanks for the feedback! Done - hostname is now auto-detected from `tailscale status --json`. Much cleaner interface now.

notheotherben committed:
style: Resolve a range of shellcheck warnings

notheotherben committed:
style: Resolve more shellcheck warnings
